### PR TITLE
Added google vision impl

### PR DESCRIPTION
**Pull request by Layoric**
Google Vision API is a SAAS for image recognition, we are only using the
`LABEL_DETECTION` to test out possible use case for indexing large
amounts of images against our custom vocabulary. Currently a very
simplistic implementation but tests proved interesting.

Could you have a look over what I've got @scelton ?

### CODE REVIEW

**Comment by jkool** [comment]
Just wanted to give you an FYI that I did get your message, and am working on getting a training image set.  I’ll get back to you when I’ve got it.  Thanks!

Johnathan

File: src/main/java/au/gov/ga/hydroid/service/impl/GoogleVisionImageService.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+         }
+      }
+      return result.substring(0,result.length() - 1);
+   }
+
+   static HttpTransport newProxyTransport(HydroidConfiguration configuration) throws GeneralSecurityException, IOException {
+      NetHttpTransport.Builder builder = new NetHttpTransport.Builder();
+      builder.trustCertificates(GoogleUtils.getCertificateTrustStore());
+      if(configuration.getProxyPort() > 0) {
+         builder.setProxy(new Proxy(Proxy.Type.HTTP, new InetSocketAddress("127.0.0.1", 3128)));
```
Comment (scelton) [inline src/main/java/au/gov/ga/hydroid/service/impl/GoogleVisionImageService.java]: Maybe the proxy settings from the configuration object.

Comment (scelton) [inline src/main/java/au/gov/ga/hydroid/service/impl/GoogleVisionImageService.java]: Other than that, all looks good.

**Comment by Layoric** [comment]
@jkool No worries :+1:

**Pull request merged**